syntax = "proto3";

package ark;

// Use add command to modify/overwrite a file
enum Command {
	ADD = 0;
	APPEND = 1;
	DELETE = 2;
	CLEAR = 3;
	READ = 4;
}

message Data {
	string file_name = 1;
	bytes value = 2;
}

message StorageRequest {
	Command command = 1;
	Data data = 2;
}

enum StorageStatus {
	SUCCESS = 0;
	FAILURE = 1;
}

message StorageResponse {
	StorageStatus status = 1;
	string file_content = 2;
}

service StorageService {
  rpc requestStorage(StorageRequest) returns (StorageResponse) {}
}
